servers/republik/modules/crowdfundings/graphql/resolvers/_mutations/appendPeriodToMembership.js

```diff
@@ -0,0 +1,66 @@
+const { Roles } = require('@orbiting/backend-modules-auth')
+const moment = require('moment')
+
+module.exports = async (_, { id, duration, durationUnit }, context) => {
+  const { user: me, t, req } = context
+  Roles.ensureUserHasRole(me, 'supporter')
+
+  const transaction = await context.pgdb.transactionBegin()
+
+  try {
+    const membership = await transaction.public.memberships.findOne({ id })
+    if (!membership) {
+      throw new Error(t('api/membership/404'))
+    }
+
+    if (duration <= 0) {
+      throw new Error(t('api/appendPeriodToMembership/durationShouldBePositive', { duration }))
+    }
+
+    const beginDate = await getBeginDate(transaction, id)
+    const endDate = getEndDate(beginDate, duration, durationUnit)
+
+    await transaction.public.membershipPeriods.insert({
+      membershipId: id,
+      beginDate,
+      endDate,
+      kind: 'ADMIN'
+    })
+
+    const updatedMembership = await transaction.public.memberships.findOne({ id })
```

Comment:
You did not alter membership (but its periods) and there is no actual need to refetch that membership. Returning initial `membership` is enough.

(A plausible `Membership.periods` query will be resolved after this mutation fetch a new period, too.)

Reply:
This makes a whole lot of sense and was my intuition on the previous PR.

As discussed with @patte, there could be some Postgres functions or trigges that actually modify the membership table. Hence it might be a good idea to always re-fetch the object. Is this ridiculous?